### PR TITLE
[SDK-3089] Support notification click callback in killed state

### PR DESCRIPTION
**Pull request by shivamsharma2710**

### CODE REVIEW

**Review by piyush-kukadiya — APPROVED** [review]
LGTM

File: android/src/main/java/com/clevertap/clevertap_plugin/CleverTapAppContextHolder.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.clevertap.clevertap_plugin;
+
+
+import android.content.Context;
+import android.util.Log;
+
+public class CleverTapAppContextHolder {
```
Comment (darshanclevertap) [inline android/src/main/java/com/clevertap/clevertap_plugin/CleverTapAppContextHolder.java:7]: @shivamsharma2710 please add javadoc on what this class does

File: android/src/main/java/com/clevertap/clevertap_plugin/CleverTapPlugin.java [hunk 2]
```diff
@@ -822,6 +841,28 @@ private Map<String, Object> getVariableValueAsMap(String name) {
                 "Variable name = " + name + " does not exist.");
     }
 
+    private void getAppLaunchNotification(Result result) {
```
Comment (darshanclevertap) [inline android/src/main/java/com/clevertap/clevertap_plugin/CleverTapPlugin.java:844]: @shivamsharma2710 add comments on what this code does

File: android/src/main/java/com/clevertap/clevertap_plugin/isolate/IsolateHandlePreferences.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import io.flutter.view.FlutterCallbackInformation;
+
+//TODO: make it singleton
```
Comment (darshanclevertap) [inline android/src/main/java/com/clevertap/clevertap_plugin/isolate/IsolateHandlePreferences.java:11]: @shivamsharma2710 Resolve the TODO or remove it

File: android/src/main/java/com/clevertap/clevertap_plugin/CleverTapAppContextHolder.java (hunk 1 above)
Comment (shivamsharma2710) [reply android/src/main/java/com/clevertap/clevertap_plugin/CleverTapAppContextHolder.java:11]: Done.

File: android/src/main/java/com/clevertap/clevertap_plugin/CleverTapPlugin.java (hunk 2 above)
Comment (shivamsharma2710) [reply android/src/main/java/com/clevertap/clevertap_plugin/CleverTapPlugin.java:852]: Resolved!

File: android/src/main/java/com/clevertap/clevertap_plugin/isolate/IsolateHandlePreferences.java (hunk 3 above)
Comment (shivamsharma2710) [reply android/src/main/java/com/clevertap/clevertap_plugin/isolate/IsolateHandlePreferences.java]: Resolved.

**Review by darshanclevertap — APPROVED** [review]

**Pull request merged**